tools/render-test/render-test-main.cpp

```diff
@@ -190,13 +190,13 @@ SlangResult _assignVarsFromLayout(
             return SLANG_E_INVALID_ARG;
         }
 
-        auto entryCursor = rootCursor.getPath(entry.name);
+        auto entryCursor = rootCursor.getPath(entry.name.getBuffer());
```

Comment:
I have a change similar to this one in my local branch, so the one thing that jumps out to me is that this version of things ends up assuming null-termination of the string for the `getPath/followPath` case, with no option for explicit termination.

The logic only works in this case because (1) our `String`s are null-terminated and (2) it isn't necessary for `render-test` to support names with embedded null bytes in them. I somehow `entry.name` got changed to something like a `UnownedStringSlice` then this code would risk breaking.

Ideally we can/should get back to a point where `ShaderCursor` can work with explicitly-terminated string slices in a convenient fashion soon (e.g., by moving the relevant bits of the `ShaderCursor` API to be inline functions in the public header).